collector/cpu_linux.go

```diff
@@ -83,8 +114,64 @@ func NewStatCollector() (Collector, error) {
 	}, nil
 }
 
+func (c *cpuCollector) Update(ch chan<- prometheus.Metric) error {
+	err1 := c.UpdateStat(ch)
+	err2 := c.UpdateCpufreq(ch)
+	if err1 != nil {
+		return err1
+	}
+	if err2 != nil {
+		return err2
+	}
```

Comment:
Oh right, you would need to check the error after each assignment. Still could use one though. Or something like:
```
var err error
for e := range []error{c.UpdateStat(ch)} {
  if e != nil {
    err = e
  }
}
```

But arguably this isn't better.. We might want to think about this case in general "partial failures in collectors".. Either way, it's also fine to merge this as is.